src/pages/node-express/lesson2-a1.mdx

```diff
@@ -0,0 +1,304 @@
+---
+layout: "../../layouts/genericMarkdownFile.astro"
+title: Node/Express Lesson 2 Assignment
+description: Node/Express Lesson 2 Assignment
+---
+
+import { ReviewLessonButton } from "../../components/navButtons.js";
+
+## **Practice with NPM and Async Patterns**
+
+<ReviewLessonButton link="/node-express/lesson2-m1" />
+
+For week 1, you created files in the `node-express-course/01-node-tutorial/answers` directory. For week 2, you'll continue to do your work in that same directory. However, before you do your work for this week, you must switch to a new git branch. While the week1 branch is active, use the command
+
+```bash
+git checkout -b week2
+```
+
+As you will be working on the `node-express-course` repository for some weeks, this is the way you separate your assignments for each week. The key topics in this section are:
+
+- how to use `npm` and the `package.json` file to manage a Node project and its dependencies
+- async patterns
+- event emitters and handlers
+- streams
+
+The instructor does provide examples if you need them. Just like the previous lesson, you will work in the `answers` directory within the `01-node-tutorial` directory. Follow these steps:
+
+1. Within your answers directory, create a file called `.gitignore`. It should have the following lines:
+
+```
+/node_modules
+.DS_Store
+```
+
+<p class="ml-8">
```

Comment:
Markdown interprets an intented block after a code block as more code; so to avoid that, I wrap such blocks of text with a `p` tag, in this case because I want it to be indented together with the numbered steps above/below.